services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -500,8 +612,81 @@ async def _clean_expired_uploads(self):
+        was initiated on S3 backend if it does not exist in file_meta_data
+        table.
+        Use-cases:
+            - presigned multipart upload: a multipart upload is created after the entry in the table (
+                if the expiry date is still in the future we do not remove the upload
+            )
+            - S3 external or internal potentially multipart upload (using S3 direct access we do not know
+            if they create multipart uploads and have no control over it, the only thing we know is the upload
+            expiry date)
+            --> we only remove dangling upload IDs which expiry date is in the past or that have no upload in process
+            or no entry at all in the database
+
+        """
+        current_multipart_uploads: list[
+            tuple[UploadID, SimcoreS3FileID]
+        ] = await get_s3_client(self.app).list_ongoing_multipart_uploads(
+            self.simcore_bucket_name
+        )
+        if not current_multipart_uploads:
+            return
+
+        # we do have some multipart uploads, let's check if they are all known to
+        # us (counterpart in file_meta_data)
+        # NOTE: S3 url encode file uuid with specific characters
+        async with self.engine.acquire() as conn:
+            list_of_known_files = await db_file_meta_data.list_fmds(
+                conn,
+                file_ids=[
+                    SimcoreS3FileID(urllib.parse.unquote(f))
+                    for _, f in current_multipart_uploads
+                ],
+            )
+        # known uploads do have an expiry date (regardless of upload ID that we do not always know)
+        list_of_known_uploads = [
+            fmd for fmd in list_of_known_files if fmd.upload_expires_at
+        ]
+        if len(current_multipart_uploads) == len(list_of_known_uploads):
+            # all good, nothing to do
+            return
+
+        # we have some "dangling" uploads here.
+        list_of_valid_upload_ids = [fmd.upload_id for fmd in list_of_known_uploads]
+        list_of_invalid_uploads = [
+            (
+                upload_id,
+                file_id,
+            )
+            for upload_id, file_id in current_multipart_uploads
+            if upload_id not in list_of_valid_upload_ids
+        ]
+        logger.debug(
+            "the following %s was found and will now be aborted",
+            f"{list_of_invalid_uploads=}",
+        )
+        await logged_gather(
+            *(
+                get_s3_client(self.app).abort_multipart_upload(
+                    self.simcore_bucket_name, file_id, upload_id
+                )
+                for upload_id, file_id in list_of_invalid_uploads
+            ),
+            max_concurrency=2,
+        )
+        logger.warning(
+            "Dangling multipart uploads %s, were aborted. "
+            "TIP: If storage was NOT improperly restarted, this might indicate that something went"
```

Comment:
yes me neither ;) as it should not happen I think... but you never know. there should be no dangling multipart upload ID. since they are cleaned it's not so bad... but still could be investigated the day we have time for that

Reply:
Maybe also add this note in either the code or the tip please.